semgrep-core/src/engine/Match_extract_mode.ml

```diff
@@ -337,7 +352,13 @@ let extract_and_concat erule_table xtarget rule_ids matches =
                   fun ({ Parse_info.charpos; _ } as loc) ->
                     if charpos < consumed_loc.start_pos then map_contents loc
                     else
-                      let line = loc.line - consumed_loc.start_line in
+                      (* For some reason, with json extraction option, it needs a fix to point the right line *)
+                      (* TODO: Find the reason of this behaviour and fix it properly *)
```

Comment:
If I had to guess, this is probably an artefact of how the JSON is formatted. Is it something like the following?
```
code: [
  "contents",
  "more contents"
]
```
(looks like yes, from line 27 in the example file below)

Relying on Yojson probably doesn't help here, because we wouldn't have an easy way to distinguish the formatting and corresponding range mapping we'd have to do.